node_ops/task_manager_module.cc

```diff
@@ -6,38 +6,128 @@
  * SPDX-License-Identifier: AGPL-3.0-or-later
  */
 
+#include "db/system_keyspace.hh"
 #include "node_ops/task_manager_module.hh"
 #include "service/storage_service.hh"
+#include "service/topology_state_machine.hh"
 #include "tasks/task_handler.hh"
 
+#include <boost/range/adaptor/transformed.hpp>
+
 namespace node_ops {
 
+static tasks::task_manager::task_state get_state(const db::system_keyspace::topology_requests_entry& entry) {
+    if (!entry.id) {
+        return tasks::task_manager::task_state::created;
+    } else if (!entry.done) {
+        return tasks::task_manager::task_state::running;
+    } else if (entry.error == "") {
+        return tasks::task_manager::task_state::done;
+    } else {
+        return tasks::task_manager::task_state::failed;
+    }
+}
+
+static std::set<tasks::task_id> get_pending_ids(service::topology& topology) {
+    std::set<tasks::task_id> ids;
+    for (auto& request : topology.requests) {
+        ids.emplace(topology.find(request.first)->second.request_id);
+    }
+    return ids;
+}
+
+static future<db::system_keyspace::topology_requests_entries> get_entries(db::system_keyspace& sys_ks, service::topology& topology) {
+    // Started requests.
+    auto entries = co_await sys_ks.get_topology_request_entries();
```

Comment:
Do we need to read entire table? It contains historical data about tasks that completed long time ago.

Reply:
The filters are added in later patches